File: src/FASTCAMPUS/ALGO/D37/P15681.js
const fs = require('fs');
class Scanner {
  constructor(fileName){
    this.read = fs.readFileSync(fileName).toString().trim().split('\n');
    this.stringToken = [];
  }
  next(){
    if(this.stringToken.length === 0){
      if(this.read.length > 0){
        this.stringToken = this.read.shift().split(" ");
      } else {
        return null;
      }
    } 
    return this.stringToken.shift();
  }

  nextNumber(){
    const next = this.next();
    if(next){
      return Number(next);
    } else {
      return null;
    }
    
  }

  nextLine(){
    if(this.read.length > 0){
      return this.read.shift();
    } else {
      return null;
    }
    
  }
}

const scanner = new Scanner('stdin.txt');
let N, R, Q;
let tree;
let parent;
let Dy;
let U
const input = function (){
  [N, R, Q] = scanner.nextLine().split(" ").map(Number);
  //console.log(N, R, Q);
  let info = scanner.read.splice(0, N -1);
  tree = Array.from({length : N + 1}, () => new Array());
  parent = Array.from({length : N + 1}, () => 0);
  Dy = Array.from({length : N + 1}, () => 1);
  info.forEach(i => {
    let [U, V] = i.split(" ").map(Number);
    tree[U].push(V);
    tree[V].push(U);
  })
  U = scanner.read.splice(0, Q).map(Number);

}
const findParent = function(vertex){
  tree[vertex].forEach(child => {
    if(parent[vertex] === child) return;
    parent[child] = vertex;
    Dy[vertex] += findParent(child);
  })
  return Dy[vertex];
}
const pro = function (){
  findParent(R);
  let answer = ""
  U.forEach(u => {
    answer += Dy[u];
    answer += '\n';
  })
  console.log(answer)
}

const solve = function(){
  input();
  pro();
}

solve();

File: src/FASTCAMPUS/ALGO/D37/P15681.java
package FASTCAMPUS.ALGO.D37;
import java.io.*;
import java.util.*;

public class P15681 {
    static int N, R, Q;
    static ArrayList<ArrayList<Integer>> tree;
    static int [] parent;
    static int [] Dy;
    static int [] U;
    static StringBuilder sb;
    static void input() {
        FastReader scan = new FastReader();
        sb = new StringBuilder();
        N = scan.nextInt();
        R = scan.nextInt();
        Q = scan.nextInt();
        tree = new ArrayList<ArrayList<Integer>>();
        for(int i = 0; i <= N; i ++){
            tree.add(new ArrayList<Integer>());
        }
        parent = new int [N + 1];
        Dy = new int [N + 1];
        for(int i = 1; i <= N; i ++){
            Dy[i] = 1;
        }
        U = new int [Q];
        for(int i = 0; i < N - 1; i ++){
            int U = scan.nextInt();
            int V = scan.nextInt();
            tree.get(U).add(V);
            tree.get(V).add(U);
        }
        //System.out.println(tree);
        for(int i = 0; i < Q; i ++){
            U[i] = scan.nextInt();
        }
    }
    public static int findParent(int vertex){
        for (int child : tree.get(vertex)) {
            if(child == parent[vertex]) continue;
            parent[child] = vertex;
            Dy[vertex] += findParent(child);
        }
        return Dy[vertex];
    }
    public static void pro(){
        findParent(R);
        
        System.out.println();
        for (int i : U) {
            sb.append(Dy[i]);
            sb.append("\n");
        }
        System.out.println(sb);
    }
    public static void main(String[] args) {
        input();
        pro();
    }
    static class FastReader {
        BufferedReader br;
        StringTokenizer st;

        public FastReader() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        public FastReader(String s) throws FileNotFoundException {
            br = new BufferedReader(new FileReader(new File(s)));
        }

        String next() {
            while (st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() {
            return Integer.parseInt(next());
        }

        long nextLong() {
            return Long.parseLong(next());
        }

        double nextDouble() {
            return Double.parseDouble(next());
        }

        String nextLine() {
            String str = "";
            try {
                str = br.readLine();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return str;
        }
    }
}